saleor/dashboard/graphql/mutations.py

```diff
@@ -31,14 +32,63 @@ def convert_form_errors(form):
 
 class BaseMutation(graphene.Mutation):
     errors = graphene.List(Error)
+    return_field_name = None
 
     class Meta:
         abstract = True
 
+    @classmethod
+    def _update_mutation_arguments_and_fields(cls, arguments, fields):
+        cls._meta.arguments.update(arguments)
+        cls._meta.fields.update(fields)
+
 
 class ModelFormMutationOptions(MutationOptions):
     form_class = None
-    return_field_name = None
+
+
+class ModelDeleteMutationOptions(MutationOptions):
+    model = None
+
+
+class ModelDeleteMutation(BaseMutation):
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def __init_subclass_with_meta__(
+            cls, arguments=None, model=None, return_field_name=None, _meta=None, **options):
+        if not model:
+            raise ImproperlyConfigured(
+                'model is required for ModelDeleteMutation')
+
+        _meta = ModelDeleteMutationOptions(cls)
+
+        if not return_field_name:
+            return_field_name = get_model_name(model)
+        if arguments is None:
+            arguments = {}
+        arguments.update({'id': graphene.ID()})
+        model_type, fields = get_model_type_and_fields(registry, model, return_field_name)
+
+        _meta.model = model
+        _meta.return_field_name = return_field_name
+        _meta.model_type = model_type
```

Comment:
Assigning `model` and `return_field_name` is here becasue it is passed as an argument and has to be used somewhere else. On the other hand, getting model type from the registry is as cheap as reading from the dictionary (look at the source code), so I think that's something that can be done on the fly in `mutate`.